lib/vectorspace.h

```diff
@@ -45,7 +45,10 @@ class VectorSpace {
  public:
   class Vector {
    public:
-    Vector(Pointer&& ptr, uint64_t size) : ptr_(std::move(ptr)), size_(size) {}
+    Vector() = delete;
+
+    Vector(Pointer&& ptr, unsigned num_qubits)
+        : ptr_(std::move(ptr)), num_qubits_(num_qubits) {}
```

Comment:
A move constructor and a move assignment operators are generated by compiler. Yes, a copy constructor and a copy assignment operators are not provided. The idea was to avoid unintentional copying of potentially very large vectors. An outside copy operation is provided for intentional copying. I have deleted a swap operation. Vectors can be swapped by std::swap.